### PR TITLE
feat(argo-cd): add support for envFrom

### PR DESCRIPTION
**Pull request by czerasz**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [ ] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/dex/deployment.yaml [hunk 1]
```diff
@@ -62,6 +62,10 @@ spec:
         {{- if .Values.dex.env }}
         env:
 {{- toYaml .Values.dex.env | nindent 8 }}
+        {{- end }}
+        {{- with .Values.dex.envFrom }}
+        envFrom:
+{{- toYaml . | nindent 8 }}
         {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/dex/deployment.yaml:69]:
```suggestion
        {{- with .Values.dex.envFrom }}
        envFrom: {{- toYaml . | nindent 8 }}
        {{- end }}
```

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 2]
```diff
@@ -70,6 +70,10 @@ spec:
         {{- if .Values.controller.env }}
         env:
 {{- toYaml .Values.controller.env | nindent 8 }}
+        {{- end }}
+        {{- with .Values.controller.envFrom }}
+        envFrom:
+{{- toYaml . | nindent 8 }}
         {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:77]:
```suggestion
        {{- with .Values.controller.envFrom }}
        envFrom: {{- toYaml . | nindent 8 }}
        {{- end }}
```

File: charts/argo-cd/templates/argocd-server/deployment.yaml [hunk 3]
```diff
@@ -67,6 +67,10 @@ spec:
         {{- if .Values.server.env }}
         env:
 {{- toYaml .Values.server.env | nindent 8 }}
+        {{- end }}
+        {{- with .Values.server.envFrom }}
+        envFrom:
+{{- toYaml . | nindent 8 }}
         {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:74]:
```suggestion
        {{- with .Values.server.envFrom }}
        envFrom: {{- toYaml . | nindent 8 }}
        {{- end }}
```

File: charts/argo-cd/templates/redis/deployment.yaml [hunk 4]
```diff
@@ -52,6 +52,10 @@ spec:
         {{- if .Values.redis.env }}
         env:
 {{- toYaml .Values.redis.env | nindent 8 }}
+        {{- end }}
+        {{- with .Values.redis.envFrom }}
+        envFrom:
+{{- toYaml . | nindent 8 }}
         {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/deployment.yaml:59]:
```suggestion
        {{- with .Values.redis.envFrom }}
        envFrom: {{- toYaml . | nindent 8 }}
        {{- end }}
```

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml [hunk 5]
```diff
@@ -67,6 +67,10 @@ spec:
           value: argocd
 {{- end }}
 {{- end }}
+        {{- with .Values.openshift.envFrom }}
+        envFrom:
+{{- toYaml . | nindent 8 }}
+        {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml:73]:
```suggestion
        {{- with .Values.openshift.envFrom }}
        envFrom: {{- toYaml . | nindent 8 }}
        {{- end }}
```

**Review by mkilchhofer — APPROVED** [review]

**Comment by czerasz** [comment]
nice, @mkilchhofer thx for the fast reaction!

**Pull request merged**